charts/gateway/templates/shared-state-client-secret.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "sharedStateClientSecretName" . }}
+  labels:
+    app: {{ template "gateway.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- range $key, $val := .Values.additionalLabels }}
+    {{ $key }}: "{{ $val }}"
+    {{- end }}
+  {{- if  .Values.additionalAnnotations }}
+  annotations:
+{{- range $key, $val := .Values.additionalAnnotations }}
+    {{ $key }}: "{{ $val }}"
+{{- end }}
+{{- end }}
+type: Opaque
+stringData:
+  sharedstate_client.yaml: |+
+    redis:
+      default:
+        commandTimeout: {{ default 5000 .Values.config.redis.commandTimeout }}
+        connectTimeout: {{ default 10000 .Values.config.redis.connectTimeout }}
+        keyPrefixGroupName: {{ default "l7GW" .Values.config.redis.groupName }}
+        testOnStart: {{  default "false" .Values.config.redis.testOnStart }}
+        {{- if .Values.config.redis.auth.enabled }}
+        {{- if .Values.config.redis.auth.username }}
+        username: {{ .Values.config.redis.auth.username | quote }}
+        {{- end }}
+        {{- if .Values.config.redis.auth.password.encoded }}
+        encodedPassword: {{ .Values.config.redis.auth.password.value | quote }}
+        {{- else }}
+        password: {{ .Values.config.redis.auth.password.value | quote }}
+        {{- end }}
+        {{- end }}
+        {{- if .Values.config.redis.sentinel.enabled }}
+        type: sentinel
+        sentinel:
+          master: {{ .Values.config.redis.sentinel.masterSet }}
+          nodes:
+          {{- if .Values.config.redis.subChart.enabled }}
+          - host: {{ .Release.Name }}-redis-node-0.{{ .Release.Name }}-redis-headless.{{ .Release.Namespace }}.svc.cluster.local
+            port: 26379
+          - host: {{ .Release.Name }}-redis-node-1.{{ .Release.Name }}-redis-headless.{{ .Release.Namespace }}.svc.cluster.local
+            port: 26379
+          - host: {{ .Release.Name }}-redis-node-2.{{ .Release.Name }}-redis-headless.{{ .Release.Namespace }}.svc.cluster.local
+            port: 26379
+          {{- else }}
+          {{- range .Values.config.redis.sentinel.nodes }}
+          - host: {{ .host }}
+            port: {{ .port }}
+          {{- end }}
+          {{- end }}
+        {{- else }}
+        type: standalone
+        standalone:
+          {{- if .Values.config.redis.subChart.enabled }}
+          host: {{ .Release.Name }}-redis-master-0.{{ .Release.Name }}-redis-headless.{{ .Release.Namespace }}.svc.cluster.local
+          port: 6379
+          {{- else }}
+          host: {{ .Values.config.redis.standalone.host }}
+          port: {{ .Values.config.redis.standalone.port }}
+          {{- end }}
+          {{- end }}
+        ssl:
+          enabled: {{ .Values.config.redis.tls.enabled }}
+          cert: redis.crt
```

Comment:
cert and verifyPeer elements only need to appear when tls is enabled

Reply:
updated